app/controllers/quizzes_controller.rb

```diff
@@ -65,8 +80,33 @@ def create
       if @quiz.save
         format.html { redirect_to quiz_url(@quiz), notice: "Quiz was successfully created." }
         format.json { render :show, status: :created, location: @quiz }
-        Student.where(teacher: current_user.email, course_id:@quiz.course_id).each do |stud|
-          Qroster.create(quiz_id:@quiz.id,student_id:stud.id)
+        if @quiz.targeted
+          recentQ = Quiz.select(:id).where(teacher:current_user.email, course_id:@quiz.course_id, completed:true).order(updated_at: :desc).limit(2).pluck(:id)
+
+          topMissed = Qroster.from(Qroster.where(quiz_id:recentQ).order(attempts: :desc)).select(:student_id).distinct.limit(10).pluck(:student_id)
+          cnt = 0
+          topMissed.each do |stud|
+            cnt = cnt + 1
+            Qroster.create(quiz_id:@quiz.id,student_id:stud)
+          end
+
+          xCnt = 0
+          if cnt == 10
+            xCnt = 5
+          elsif cnt <= 15
```

Comment:
would it every reach to 15 if limit is 10?

Reply:
I don't think so, I could change it to <= 10. We can tweak this block. Main idea is just to determine how many extra random students we want to mix in

Reply:
Since we know that the limit, can you just do `xCnt = 15-cnt` instead of the whole block?

Reply:
@jkelly8823 looks like you forgot to remove this.

Reply:
I'm fixing that and adding the nil checking, will push in a few minutes